lib/train/platforms/detect/specifications/os.rb

```diff
@@ -17,9 +17,17 @@ def self.load
 
       plat.family('windows').in_family('os')
           .detect {
-            if winrm? || (@backend.local? && ruby_host_os(/mswin|mingw32|windows/))
-              true
+            # Can't return from a `proc` thus the `is_windows` shenanigans
+            is_windows = false
+            is_windows = true if winrm?
+
+            if defined?(Train::Transports::Local::Connection)
+              if @backend.class == Train::Transports::Local::Connection
```

Comment:
You still need to check for `ruby_host_os(/mswin|mingw32|windows/)` here

Reply:
Great catch will update.